src/test/java/com/commercetools/emailprocessor/email/EmailProcessorTest.java

```diff
@@ -0,0 +1,223 @@
+package com.commercetools.emailprocessor.email;
+
+import com.commercetools.emailprocessor.model.Email;
+import com.commercetools.emailprocessor.model.Statistics;
+import com.commercetools.emailprocessor.model.TenantConfiguration;
+import io.sphere.sdk.client.SphereClient;
+import io.sphere.sdk.customobjects.CustomObject;
+import io.sphere.sdk.customobjects.queries.CustomObjectQuery;
+import io.sphere.sdk.queries.PagedQueryResult;
+import org.apache.commons.io.IOUtils;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.*;
+import java.net.*;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+import static org.junit.Assert.*;
+
+
+public class EmailProcessorTest {
+
+
+    public EmailProcessor emailProcessor = null;
+    TenantConfiguration tenantConfiguration = null;
+    List<CustomObject<Email>> customObjects = null;
+    private static final Logger LOG = LoggerFactory.getLogger(EmailProcessor.class);
+
+    @Before
+    public void setUp() throws Exception {
+        customObjects = new ArrayList<CustomObject<Email>>();
+        emailProcessor = Mockito.mock(EmailProcessor.class);
+        Mockito.doCallRealMethod().when(emailProcessor).processEmails(Mockito.any(TenantConfiguration.class));
+        tenantConfiguration = new TenantConfiguration();
+        tenantConfiguration.setProjectKey("testproject");
+        //tenantConfiguration.setClient(mockSphereClient());
+    }
+
+    @Test
+    public void shouldProcessEmails() throws Exception {
+        customObjects.add(createCustomObject("1", "pending", Statistics.RESPONSE_CODE_SUCCESS));
+        customObjects.add(createCustomObject("2", "pending", Statistics.RESPONSE_ERROR_TEMP));
+        customObjects.add(createCustomObject("3", "error", Statistics.RESPONSE_ERROR_TEMP));
+        customObjects.add(createCustomObject("4", "pending", Statistics.RESPONSE_ERROR_PERMANENT));
+        tenantConfiguration.setClient(mockSphereClient(customObjects));
+        Statistics statistic = emailProcessor.processEmails(tenantConfiguration);
+        assertEquals(statistic.getProcessedEmail(), 3);
+        assertEquals(statistic.getSuccessful(), 1);
+        assertEquals(statistic.getTempError(), 1);
+        assertEquals(statistic.getPermanentError(), 1);
+    }
+
+    @Test
+    public void shouldProcessEmails2() throws Exception {
```

Comment:
Why do you need `shouldProcessEmails2`? Why `shouldProcessEmails` is not enough?

Reply:
What I meant was what is the difference between `shouldProcessEmails` and `shouldProcessEmails2`? Now you group them into 1 method, but I don't see any real benefit or `shouldProcessEmails2`?

Reply:
b7b2ba68262bbdd2f0be45f2f06bad1b1861b4fe